V2 Beta of frontend

Yantrio opened the pull request:
This PR sits on top of #338 

It does the following:
- Unify the search bar components, as they were all a mess!
- Make the landing page much less jarring by not altering the UI when you hit the search bar

## Checklist

- [ ] I have read the [contribution guide](https://github.com/opentofu/opentofu/blob/main/CONTRIBUTING.md).
- [ ] I have not used an AI coding assistant to create this PR.
- [ ] My contribution is compatible with the MPL-2.0 license and I have provided a DCO sign-off on all my commits.
- [ ] I have written all code in this PR myself OR I have marked all code I have not written myself (including modified code, e.g. copied from other places and then modified) with a comment indicating where it came from.

yottta reviewed (approved):
Feels way better. I am not sure, but feels that something is still missing. Maybe others have a more valuable feedback.

Gogotchuri commented:
The old homepage was better—more clarity and a simple UX, featuring a single search bar with a good call to action.
The main issue was that the jump to a different layout occurred as soon as you started searching, which was jarring and unexpected.
One thing I can think of is leaving the Home page as it was before and changing the search bar so that it displays the autocomplete list of resources. We can then jump to the resource once the user clicks or presses Enter. This is the intended action and shouldn't be an issue.
The corner search bar can be reused here.
<img width="892" height="818" alt="image" src="https://github.com/user-attachments/assets/17c9553a-c8ce-4fe1-bf94-3d03329b66ca" />

P.S. Another thing I have overlooked - We should remove that corner search bar entirely. Having two search bars on pages is a bit confusing and unnecessary.

Yantrio committed:
Search experience and UI rework

Squashed 30 WIP commits covering:
- Search experience rework and modernized searchbar
- Modernized provider and module pages to match OpenTofu.org aesthetic
- Breadcrumbs rework with aria-current support
- Cleaned up docs, version, and provider/module rendering
- ToC sidebar, anchor scrolling improvements
- License rendering cleanup
- Landing page rework
- Various accessibility and styling fixes

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
Fixed egregous linting issues

Signed-off-by: James Humphries <james@james-humphries.co.uk>

Yantrio committed:
Fixed linting

Signed-off-by: James Humphries <james@james-humphries.co.uk>
